### PR TITLE
Fix error emitting

### LINKED ISSUE #17: Error not emitted if file cannot be open
We have a problem with logger dependent on `WritableFileStream`. For example `EACCES` will not be emitted on logger if file cannot be open.

### PR DESCRIPTION
**Pull request by bugagashenkj**
Close #17

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "eslint-config-prettier": "^3.6.0",
```
Comment (lundibundi) [inline package.json]: Why did you update metatests in this PR?

File: lib/fs-writable.js [hunk 2]
```diff
@@ -75,10 +75,11 @@ class WritableFileStream extends Writable {
 
   [errorSymbol](error, callback) {
     if (callback) callback(error);
+    this.emit('error', error);
+
     if (this.autoClose) {
       this.destroy(error);
```
Comment (lundibundi) [inline lib/fs-writable.js]: Wouldn't the 'error' be emitted twice if `autoClose` is on?
Relevant `destroy` code
https://github.com/metarhia/metastreams/blob/1622a2816305e03577e06300fcf7e7db09d4a2fa/lib/writable.js#L218-L224

File: test/fs-writable.js [hunk 3]
```diff
@@ -45,6 +45,15 @@ metatests.test('fs-writable / open', test => {
   test.on('done', () => fs.unlinkSync(TEST_FILENAME));
 });
 
+metatests.test('fs-writable / open with error', test => {
+  const TEST_FILENAME = path.join(__dirname, '.');
```
Comment (SemenchenkoVitaliy) [inline test/fs-writable.js]: Why `path.join(__dirname, '.')`? The result will be just `__dirname`.

File: test/fs-writable.js [hunk 4]
```diff
@@ -45,6 +45,15 @@ metatests.test('fs-writable / open', test => {
   test.on('done', () => fs.unlinkSync(TEST_FILENAME));
 });
 
+metatests.test('fs-writable / open with error', test => {
+  const TEST_FILENAME = __dirname;
```
Comment (lundibundi) [inline test/fs-writable.js]:
```suggestion
  const INVALID_TEST_FILENAME = __dirname;
```

File: test/fs-writable.js [hunk 5]
```diff
@@ -45,6 +45,15 @@ metatests.test('fs-writable / open', test => {
...
+
+  test.plan(1);
+
+  const stream = new WritableFileStream(TEST_FILENAME);
+  stream.on('error', () => test.pass('open'));
```
Comment (lundibundi) [inline test/fs-writable.js]: I'd prefer to explicitly check error that was emitted. (to avoid this test passing when some other error is emitted)

Comment (lundibundi) [inline test/fs-writable.js]: Nit: you can use
```suggestion
  stream.on('error', test.mustCall(() => test.end()));
```
Instead od `test.plan`.

**Comment by lundibundi** [comment]
Landed in 186e694d75c4dd004b8e33045f7e0b28a480e986 and  a6390586d042d599f1198288203bf61697bd0434.

**Comment by tshemsedinov** [comment]
Please publish to npm

**Pull request merged**